ten-simple-rules-dockerfiles.Rmd

```diff
@@ -638,69 +603,53 @@ This variety of approaches render seemingly more convenient uncontainerised envi
 This habit can be supported by using continuous deployment or CI strategies.
-If a VCS repository with a `Dockerfile` is linked to an automated build, then a CI pipeline can build images and execute a validating run the new image, e.g., using contained demo data (see&nbsp;\ruleref{rule:mount}).
 
 In the case of needing setup or configuration for the first two habits, it is good practice to provide a `Makefile` alongside your container, which can capture the specific commands.
-The effective use of a `Makefile` can help avoiding undocumented steps or manual, extra commands to be run on the local machine.
-A fully scripted configuration makes it easier for both you and future users, and can increase trust in your workflow.
-If you use a `Makefile`, consider pointing to it instead of adding usage instructions in a comment at the end of the `Dockerfile`.
-A `Makefile` will also help you to take advantage of build-time variables in a useful way (unlike \ruleref{rule:favour}) for inserting specific metadata into an image, e.g., the current VCS [commit hash]() or a timestamp.
 
 Third, from time to time you can reduce the system resources occupied by Docker images and their layers or unused containers, volumes and networks by running `docker system prune --all`.
 After a prune is performed, it follows naturally to rebuild a container for local usage, or to pull it again from a newly built registry image.
 This habit can be automated with a cron job [@wikipedia_contributors_cron_2019].
 
 <!-- [SJE:  "sure, okay, I'd like to export an image.  Give me a recipe to show me how to do it, e.g. for this paper!"] -->
 Fourth, you can export the image to file and deposit it in a public data repository, where it not only becomes citable but also provides a snapshot of the _actual_ environment you used at a specific point in time.
-You should include instructions how to import and run the workflow based on the image archive and add your own image tags for clarity.
+You should include instructions for how to import and run the workflow based on the image archive and add your own image tags for clarity.
 Depositing the image next to other project files, i.e., data, code, and the used `Dockerfile`, in a public repository makes them likely to be preserved, but is is highly unlikely that over time you will be able to recreate it precisely from the accompanying `Dockerfile`.
 Publishing the image and the contained metadata therein (e.g., the Docker version used) may even allow future science historians to emulate the Docker runtime environment.
-Applying proper preservation strategies (cf. [@emsley_framework_2018]) can be highly complex, but simply running an image "as-is", i.e. with the default command and entrypoint (see \ruleref{rule:interactive}), and observing the output is quite likely to work for many years into the future.
```

Comment:
Another one of my darlings :skull:  !! I agree though that container preservation as a topic is not mature enough yet to expose the target audience of this article.